src/Microsoft.Diagnostics.Runtime/src/Utilities/Platform/LinuxFunctions.cs

```diff
@@ -9,6 +9,59 @@ namespace Microsoft.Diagnostics.Runtime
 {
     internal sealed class LinuxFunctions : PlatformFunctions
     {
+        private const string LibDlGlibc = "libdl.so.2";
+        private const string LibDl = "libdl.so";
+
+        private readonly Func<string, IntPtr> _loadLibrary;
+        private readonly Func<IntPtr, bool> _freeLibrary;
+        private readonly Func<IntPtr, string, IntPtr> _getExport;
+
+        public LinuxFunctions()
+        {
+            Type nativeLibraryType = Type.GetType("System.Runtime.InteropServices.NativeLibrary, System.Runtime.InteropServices", throwOnError: false);
+            if (nativeLibraryType != null)
+            {
+                // .NET Core 3.0+
+                _loadLibrary = (Func<string, IntPtr>)nativeLibraryType.GetMethod("Load", new Type[] { typeof(string) })?.CreateDelegate(typeof(Func<string, IntPtr>));
```

Comment:
`_loadLibrary`, `freeLibrary`, `_getExport` can all be `null` if the `GetMethod()` calls fail.

Reply:
Why would it fail? The `NativeLibrary` methods we're using are part of the public corefx API.

Reply:
1.  Null checks are cheap.
2.  The coreclr runtime could change in the future.
3.  What if someone tries to run it on a runtime which has a different implementation?  (Mono.)
4.  It's always better to defensively code if it doesn't cost anything.

Reply:
> Why would it fail? The `NativeLibrary` methods we're using are part of the public corefx API.

Since you use `?.`, you expect the calls to sometimes fail. Public APIs can change and be removed in future versions, even if it doesn't happen very often.

Reply:
I've added the null checks.